### DISCUSSION TITLE
Help me understand performance impact on the app

### DISCUSSION THREAD

**Discussion by shwarcu** [faker-js/faker#970] [Q&A] [answered]
Hello 👋 
I am experimenting with Faker for generating test data for integration and performance tests.
When I run profiling on the app that has included script that generates data I see that there is quite long-lasting scripting task that looks like faker-js initialization. 
This part takes quite a lot of time, considering that generating 5,000 test objects took around 15 ms.
<img width="1721" alt="FakerJS in profiling view" src="https://user-images.githubusercontent.com/58426925/169301004-8c73b397-b480-4d56-980f-53cb2c43d0eb.png">

Am I correct thinking that this long 221ms bar is faker-js initialization? Is there a way to improve this performance? maybe by only loading parts of the faker that I am using?

**Comment by Shinigami92** [comment, marked as answer]
Hey @shwarcu 
Some of us are already planning to optimize some stuff like e.g. splitting Faker into MinimalFaker, BaseFaker and Faker itself. (https://github.com/faker-js/faker/discussions/805#discussioncomment-2690233)

I'm not sure what's exactly going on, and I also assume if it is not easy to fix we wont work on it right now.
IMO generating 5.000 test objects within 15ms sounds fast :thinking: at least fast enough.

I'm a bit confused about that it is especially the faker-error but I don't have knowledge about profiling at all.
I think the best thing would be if you could just checkout the repo and try to debug and improve stuff on your own and if you found something create a PR for that.

**Reply by shwarcu** [reply]
Thanks for the reply! Perhaps I will take a look at the source and try to debug the performance,  my javascript skills aren't huge tho.

The good part of the problem seems to be that this long ruining process happens only once,  when library is used for the first time.  If I find anything I'll share

**Reply by shwarcu** [reply]
@Shinigami92 I'm not sure if you understand me 100% so just want to make sure

> IMO generating 5.000 test objects within 15ms sounds fast 🤔 at least fast enough.

yes this is very fast and I am totally satisfied with such result. I mentioned 15ms in order to compare it to quite long-lasting initialisation time of the library (probably this is what's happening) which seems to take around 221ms

**Reply by ST-DDT** [reply]
Have you tried the single locale variants?
E.g. `@faker-js/faker/locale/de`?

**Reply by shwarcu** [reply]
No, I will give it a try!

**Reply by shwarcu** [reply]
@ST-DDT I confirmed that importing single locale variant is significantly faster ~49-34 ms ✅
